Fix "static asset not found" error in  doNotSaveLoginFromPromptTest, saveLoginFromPromptTest UI tests

Passes locally with most recent A-C snapshot (.gradle cleared).

Fails on Firebase

Disabling these 2 tests for now to unblock master because these are clearly not UI failures.  
Most likely this issue is isolated to one of these 2 tests.
 doNotSaveLoginFromPromptTest, saveLoginFromPromptTest
in SettingsPrivacyTest

@AaronMT @rpappalax to investigate

So this is reproducible for me locally too, (and on x86 AVD runs). The file exists because I replace this line:

```Kotlin
        val url = server.url("pages/password.html").toString().toUri()!!
```

with 

```Kotlin
        val url = server.url("pages/passwordsubmit.html").toString().toUri()!!
```

and I see the page on test-execution, so the asset exists, it must be related to asset serving in MockWebServer again.

https://console.firebase.google.com/u/0/project/moz-fenix/testlab/histories/bh.66b7091e15d53d45/matrices/4651621488462066564

[Raw log-cat](https://00e9e64bac32d15b5754106392996d6b084b34b92dccad225a-apidata.googleusercontent.com/download/storage/v1/b/fenix_test_artifacts/o/2020-03-25_15-49-25.797000_tfnC%2Fshard_29%2FNexus6P-27-en-portrait%2Flogcat?qk=AD5uMEtG7l-UPfql89wiNhAsPZcsy-YvTj8c1ALwvi8ItLqPAzSuS8cNSiT3UZBRyRRITTRiyVlQWLy32zI_NgoELR3-lpvXfo2UI78GQ17D0Pqj8GSiOVQXFffwduklNQVLbRaWQDGq8M9_dTYO20_bEsdeWJCFBYS5-1XhSLAY9WNsvfA9-CmqNg08Sql9AF9G8frIFIJB5gsi8_49Lj4w5S2qmEwu2LKy7AAl6wwWVkCPKH53-tieZwMImgkA3nBeq5NfplHtKOBARee7Iarjj1eE8iLzMmsTxqcW0N1Zb4FNDyi-WbGwTsYATYJaZQfxINioHPwyrJJe83pWZs044HaUVkneyGEUQqamuo8zRGnMOQPbtrTA5bQPeKVYedWWwE99k2iJLUBL_Cl1ca99vBffLUmbbZ_W-Tj1oneuhLy7N6ic6_NV1XwYHwGKpFRS7cOnuk4tFZR8KnjIMbFWvX2-pGkfGQ166lndnnDbC8DJz3mFL7piEorasusH3MqVe1yPH4YXOnEUUPV3ob_340VpJvAcoSw_4Lwh0tbjBwuzEPMPElBakcj_ahApvYMNYRGKbgT_mG5iIHhOFVqQST4bE84UY5tkUZMwolCFY5uxpCHs7q8pSxgtsa0ArU6CvOdWkjbY73LqCgrdd_D2FysN4W1Tt3bEGhg_SDEhdipY9Jxug98JS1R-JqibbCFgohXEPWtR1kLxKbJ_y0O0EjRvAsh6GUAyWaxcqCJJcfJu0QNdpnzIrv9M84nVwqVkEE93IQFWktYiOtLnpMVSSNBLC-XtWxucfBYhvT5QtIDVQHodvT-PnhzUJ6KtHfJ4j5Ic79HU8AlV5EtGZBohC58aUZ5RtxBpOvYnWeXO-Jjt8ELQ_ORxejjSDa-dhHZVhAlf5SCt&isca=1)

```App crashed during testing```

```03-25 11:52:47.773: D/GeckoSession(7555): handleMessage GeckoView:PageStart uri=http://localhost:42419/pages/passwordsubmit.html?
03-25 11:52:47.879: I/MockWebServer(7555): MockWebServer[42419] received request: GET /pages/passwordsubmit.html? HTTP/1.1 and responded: HTTP/1.1 404 Client Error
```

```Log
   FATAL EXCEPTION: main
Process: org.mozilla.fenix.debug, PID: 7769
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:503)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
Caused by: java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	... 1 more
Caused by: java.io.FileNotFoundException: pages/passwordsubmit.html?
	at android.content.res.AssetManager.nativeOpenAsset(Native Method)
	at android.content.res.AssetManager.open(AssetManager.java:744)
	at android.content.res.AssetManager.open(AssetManager.java:721)
	at org.mozilla.fenix.helpers.AndroidAssetDispatcher.dispatch(MockWebServer.kt:69)
```


```Log
03-25 13:22:49.769: E/TestRunner(8112): failed: doNotSaveLoginFromPromptTest(org.mozilla.fenix.ui.SettingsPrivacyTest)
03-25 13:22:49.769: E/TestRunner(8112): ----- begin exception -----
03-25 13:22:49.780: E/TestRunner(8112): java.lang.RuntimeException: Unknown error during interactions
03-25 13:22:49.780: E/TestRunner(8112): 	at androidx.test.espresso.InteractionResultsHandler.finalResult(InteractionResultsHandler.java:109)
03-25 13:22:49.780: E/TestRunner(8112): 	at androidx.test.espresso.InteractionResultsHandler.gatherAnyResult(InteractionResultsHandler.java:81)
03-25 13:22:49.780: E/TestRunner(8112): 	at androidx.test.espresso.InteractionResultsHandler.gatherAnyResult(InteractionResultsHandler.java:52)
03-25 13:22:49.780: E/TestRunner(8112): 	at androidx.test.espresso.ViewInteraction.waitForAndHandleInteractionResults(ViewInteraction.java:320)
03-25 13:22:49.780: E/TestRunner(8112): 	at androidx.test.espresso.ViewInteraction.desugaredPerform(ViewInteraction.java:178)
03-25 13:22:49.780: E/TestRunner(8112): 	at androidx.test.espresso.ViewInteraction.perform(ViewInteraction.java:119)
03-25 13:22:49.780: E/TestRunner(8112): 	at org.mozilla.fenix.helpers.ViewInteractionKt.click(ViewInteraction.kt:11)
03-25 13:22:49.780: E/TestRunner(8112): 	at org.mozilla.fenix.ui.robots.BrowserRobot$Transition.openHomeScreen(BrowserRobot.kt:321)
03-25 13:22:49.780: E/TestRunner(8112): 	at org.mozilla.fenix.ui.SettingsPrivacyTest.doNotSaveLoginFromPromptTest(SettingsPrivacyTest.kt:177)
03-25 13:22:49.780: E/TestRunner(8112): 	at java.lang.reflect.Method.invoke(Native Method)
03-25 13:22:49.780: E/TestRunner(8112): 	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
```

--- a/app/src/androidTest/java/org/mozilla/fenix/helpers/MockWebServer.kt
+++ b/app/src/androidTest/java/org/mozilla/fenix/helpers/MockWebServer.kt
@@ -62,12 +62,12 @@ const val HTTP_NOT_FOUND = 404
 class AndroidAssetDispatcher : Dispatcher() {
     private val mainThreadHandler = Handler(Looper.getMainLooper())
 
-    override fun dispatch(request: RecordedRequest): MockResponse {
+    override fun dispatch(request: RecordedRequest?): MockResponse {
         val assetManager = InstrumentationRegistry.getInstrumentation().context.assets
         try {
-            val pathNoLeadingSlash = request.path.drop(1)
-            assetManager.open(pathNoLeadingSlash).use { inputStream ->
-                return fileToResponse(pathNoLeadingSlash, inputStream)
+            val pathWithoutQueryParams = Uri.parse(request?.path?.drop(1)).path
+            assetManager.open(pathWithoutQueryParams!!).use { inputStream ->
+                return fileToResponse(pathWithoutQueryParams, inputStream)
             }
         } catch (e: IOException) { // e.g. file not found.
             // We're on a background thread so we need to forward the exception to the main thread.

--- a/app/src/androidTest/java/org/mozilla/fenix/ui/SettingsPrivacyTest.kt
+++ b/app/src/androidTest/java/org/mozilla/fenix/ui/SettingsPrivacyTest.kt
@@ -141,7 +141,6 @@ class SettingsPrivacyTest {
     }
 
     @Test
-    @Ignore("Passes locally, fails on CI. Fix in https://github.com/mozilla-mobile/fenix/issues/9189")
     fun saveLoginFromPromptTest() {
         val saveLoginTest =
             TestAssetHelper.getSaveLoginAsset(mockWebServer)
@@ -167,7 +166,6 @@ class SettingsPrivacyTest {
     }
 
     @Test
-    @Ignore("Passes locally, fails on CI. Fix in https://github.com/mozilla-mobile/fenix/issues/9189")
     fun doNotSaveLoginFromPromptTest() {
         val saveLoginTest = TestAssetHelper.getSaveLoginAsset(mockWebServer)
 

